### ISSUE TITLE
Simple description environment fails to parse

### ISSUE THREAD

**Issue by brettviren** [alvinwan/TexSoup#43]
This simple example fails using 0.1.3 from PyPI on Python 3.6
```
In [2]: TexSoup.TexSoup(r'\begin{description}\item[aaa] BBB\end{description}')
```
Traceback is below.

This patch seems to avoid the failure but I don't know if it has some bad side effect.

```diff
--- venv3/lib/python3.6/site-packages/TexSoup/utils.py~	2018-08-18 11:11:00.860261440 -0400
+++ venv3/lib/python3.6/site-packages/TexSoup/utils.py	2018-08-18 11:46:32.860889811 -0400
@@ -144,6 +144,7 @@
 
     @classmethod
     def join(cls, tokens, glue=''):
+        tokens = [t for t in tokens if t]
         if len(tokens) > 0:
             return TokenWithPosition(glue.join(t.text for t in tokens),
                                      tokens[0].position)
```
After applying

```
In [2]: TexSoup.TexSoup(r'\begin{description}\item[aaa] BBB\end{description}')
Out[2]: \begin{description}\item [aaa] BBB\end{description}
```

Here's full session with traceback

```
$ ipython
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import TexSoup

In [2]: TexSoup.TexSoup(r'\begin{description}\item[aaa] BBB\end{description}')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-ca4e34c553a0> in <module>()
----> 1 TexSoup.TexSoup(r'\begin{description}\item[aaa] BBB\end{description}')

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/__init__.py in TexSoup(tex)
     82     SOUP
     83     """
---> 84     parsed, src = read(tex)
     85     return TexNode(parsed, src=src)

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/tex.py in read(tex)
     17     buf, children = Buffer(tokenize(tex)), []
     18     while buf.hasNext():
---> 19         content = read_tex(buf)
     20         if content is not None:
     21             children.append(content)

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/reader.py in read_tex(src)
    244             src.backward(candidate_index)
    245         if mode == 'begin':
--> 246             read_env(src, expr)
    247         return expr
    248     if c in ARG_START_TOKENS:

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/reader.py in read_env(src, expr)
    317         contents = [src.forward_until(lambda s: s == '\\end')]
    318     while src.hasNext() and not src.startswith('\\end{%s}' % expr.name):
--> 319         contents.append(read_tex(src))
    320     if not src.startswith('\\end{%s}' % expr.name):
    321         end = src.peek((0, 5))

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/reader.py in read_tex(src)
    227         command = TokenWithPosition(c[1:], src.position)
    228         if command == 'item':
--> 229             extra = read_item(src)
    230             mode, expr = 'command', TexCmd(command, (), extra)
    231         elif command == 'begin':

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/reader.py in read_item(src)
    267         return not any([s.startswith(substr) for substr in string.whitespace])
    268 
--> 269     last = stringify(src.forward_until(criterion))
    270     if last.startswith(' '):
    271         last = last[1:]

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/reader.py in <lambda>(s)
    261     :return: contents of the item
    262     """
--> 263     stringify = lambda s: TokenWithPosition.join(s.split(' '), glue=' ')
    264 
    265     def criterion(s):

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/utils.py in join(cls, tokens, glue)
    146     def join(cls, tokens, glue=''):
    147         if len(tokens) > 0:
--> 148             return TokenWithPosition(glue.join(t.text for t in tokens),
    149                                      tokens[0].position)
    150         else:

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/utils.py in <genexpr>(.0)
    146     def join(cls, tokens, glue=''):
    147         if len(tokens) > 0:
--> 148             return TokenWithPosition(glue.join(t.text for t in tokens),
    149                                      tokens[0].position)
    150         else:

AttributeError: 'str' object has no attribute 'text'
```

**Comment by SimonMaenaut** [comment]
Note: This issue was fixed by #34 .

```
simon@zap:~/TexSoup$ git checkout 16716b2166092e49e264a923e95b95be131ce102
Note: checking out '16716b2166092e49e264a923e95b95be131ce102'.
…
HEAD is now at 16716b2 Fix items with labels (#34)
simon@zap:~/TexSoup$ python3.6
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from TexSoup import TexSoup
>>> tex_soup = TexSoup(r'\begin{description}\item[aaa] BBB\end{description}')
>>> list(tex_soup.contents)
[\begin{description}\item[aaa] BBB\end{description}]
>>> list(tex_soup.description.contents)
[\item[aaa] BBB]
>>> list(tex_soup.description.item.args)
[OArg('aaa')]
>>> type(tex_soup.description.item.args)
<class 'TexSoup.data.TexArgs'>
>>> type(tex_soup.description.item.args(0))
<class 'TexSoup.data.OArg'>
>>> list(tex_soup.description.item.extra)
['BBB']
>>> type(tex_soup.description.item.extra[0])
<class 'TexSoup.utils.TokenWithPosition'>
>>> exit()
simon@zap:~/TexSoup$ 
```

**Comment by brettviren** [comment]
Thanks, sorry for not checking the issues closely enough.  I confirm installing from `master` fixes this.
